staging/src/k8s.io/apiserver/pkg/storage/etcd3/etcdfeature/feature_support_checker_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdfeature
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	fgt "k8s.io/component-base/featuregate/testing"
+)
+
+// MockEtcdClient is a mock implementation of the EtcdClientInterface interface.
+type MockEtcdClient struct {
+	Version string
+	// Indicates whether to return an error on Status
+	ReturnError  bool
+	ErrorMessage error
+}
+
+func (m *MockEtcdClient) Endpoints() []string {
+	return []string{"localhost:2390"}
+}
+
+// Status returns a mock status response.
+func (m *MockEtcdClient) Status(ctx context.Context, endpoint string) (*clientv3.StatusResponse, error) {
+	if m.ReturnError {
+		return nil, m.ErrorMessage
+	}
+	// Return a mock status response
+	return &clientv3.StatusResponse{
+		Version: m.Version,
+	}, nil
+}
+
+func TestCheckIsSupportedRequestWatchProgress(t *testing.T) {
+	tests := []struct {
+		name                  string
+		mockClientVersion     string
+		mockClientEndpoint    string
```

Comment:
You don't use other values of `mockClientEndpoint` than "localhost:2380". No need to parameterize on it.
```suggestion
```